modules/deployment-scripts/create-agw-kv-certificate/main.bicep

```diff
@@ -0,0 +1,229 @@
+@description('Managed identity requires "Managed Identity Operator" permission over the user assigned identity of the Application Gateway.')
+resource agwIdRbac 'Microsoft.Authorization/roleAssignments@2020-08-01-preview' =  if (!empty(agwName) && agwCertType != 'none') {
+  scope: agwId
+  name: guid(agwId.id, managedIdentityOperator, useExistingManagedIdentity ? existingDepScriptId.id : newDepScriptId.id)
+  properties: {
+    roleDefinitionId: managedIdentityOperator
+    principalId: useExistingManagedIdentity ? existingDepScriptId.properties.principalId : newDepScriptId.properties.principalId
+    principalType: 'ServicePrincipal'
+  }
+}
+
+@batchSize(1)
+resource createImportCert 'Microsoft.Resources/deploymentScripts@2020-10-01' = [for certName in certNames: {
+  name: 'AKV-Cert-${akv.name}-${replace(replace(certName,':',''),'/','-')}'
+  location: location
+  identity: {
+    type: 'UserAssigned'
+    userAssignedIdentities: {
+      '${useExistingManagedIdentity ? existingDepScriptId.id : newDepScriptId.id}': {}
+    }
+  }
+  kind: 'AzureCLI'
+  dependsOn: [
+    rbacKv
+    rbacAppGw
+  ]
+  properties: {
+    forceUpdateTag: forceUpdateTag
+    azCliVersion: azCliVersion
+    timeout: timeout
+    retentionInterval: retention
+    environmentVariables: [
+      {
+        name: 'RG'
+        value: resourceGroup().name
+      }
+      {
+        name: 'akvName'
+        value: akvName
+      }
+      {
+        name: 'agwName'
+        value: agwName
+      }
+      {
+        name: 'agwCertType'
+        value: agwCertType
+      }
+      {
+        name: 'certName'
+        value: certName
+      }
+      {
+        name: 'initialDelay'
+        value: initialScriptDelay
+      }
+      {
+        name: 'certificateWait'
+        value: agwCertImportDelay
+      }
+    ]
+    scriptContent: '''
+      #!/bin/bash
+      set -e
+
+      echo "Waiting on Identity RBAC replication ($initialDelay)"
+      sleep $initialDelay
+
+      echo "Creating AKV Cert $certName"
+      az keyvault certificate create --vault-name $akvName -n $certName -p "$(az keyvault certificate get-default-policy | sed -e s/CN=CLIGetDefaultPolicy/CN=${certName}/g )";
+
+      echo "Waiting for AKV Cert Generation ($certificateWait)"
+      sleep $certificateWait
+
+      echo "getting akv secretid for $certName to add to $agwName";
+      versionedSecretId=$(az keyvault certificate show -n $certName --vault-name $akvName --query "sid" -o tsv);
+      unversionedSecretId=$(echo $versionedSecretId | cut -d'/' -f-5) # remove the version from the url;
+
+      case $agwCertType in
+        none)
+          echo "AppGw Certificate Creation not configured"
+          ;;
+
+        root-cert)
+          echo "Creating root certificate in application gateway";
+          rootcertcmd="az network application-gateway root-cert create --gateway-name $agwName -g $RG -n $certName --key-vault-secret-id $unversionedSecretId";
```

Comment:
adding the cert is a PUT as I understand it, which means this should be done outside of the script...

Reply:
agreed